src/Textarea.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import cx from 'classnames';
+import autosize from 'autosize';
+
+/**
+ * @module Textarea
+ */
+class Textarea extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value || '',
+		};
+		this.onChange = this.onChange.bind(this);
+	}
+
+	onChange(e) {
+		this.setState({
```

Comment:
I think the `TextInput` also has an optional `onChange` prop that can be passed in so that the parent can be notified of changes - I'd recommend going ahead and adding that here

```js
onChange(e) {
  const { onChange } = this.props;
  if (onChange) {
    onChange(e);
  }
  this.setState({ value: e.target.value });
}
```